### ISSUE TITLE
Fetching Keys fails during signature validation with context canceled

### ISSUE THREAD

**Issue by tstoermer** [hashicorp/vault-plugin-auth-jwt#11]
We are using the auth/jwt to authenticate with Vault. It is configured using a discovery url. When calling the endpoint e.g.:
`vault write auth/jwt/login role=default jwt=@myToken.txt`
It somehow stops working with the failure:
```
Error writing data to auth/retest/login: Error making API request.

URL: PUT https://127.0.0.1:8200/v1/auth/jwt/login
Code: 400. Errors:

* error validating signature: failed to verify signature: fetching keys oidc: get keys failed Get https://myauthserver:8999/auth/realms/master/protocol/openid-connect/certs: context canceled
```
After configuring the discovery URL again, the issue is solved. The issue reoccurs after some time (several hours).

There was no change at the authorization server providing the tokens as described in #10  But maybe the issues are related. I also found hashicorp/vault-plugin-auth-gcp#43 what seems to be similar.

Maybe an already canceled context is forwarded to the verifier (go-oidc) using it to fetch the keys?
https://github.com/hashicorp/vault-plugin-auth-jwt/blob/fb9c940ad0ebccbd4dcaeed816bc1e6e6a4744f3/path_login.go#L142

Vault 0.10.4
Keycloak 3.4.3 & 4.3.0

**Comment by avoidik** [comment]
as a workaround you may set `oidc_discovery_ca_pem` explicitly

**Comment by MartinLeedotOrg** [comment]
My related issue was resolved somehow in Vault 0.11.1 - I haven't checked to see exactly how/why. Might be worth trying.

**Comment by tstoermer** [comment]
Thanks for your hint - I will plan to update the vault server. 

The strange thing: over the last days the error did not reoccur... So I am currently not able to reproduce it and don't know what triggered the error. 
I will continue testing this week and close the issue if it no longer exists.
Thanks all for you help (@avoidik, @MartinLeedotOrg)

**Comment by tstoermer** [comment]
Closed - not able to reproduce / did not reoccur until now.

**Comment by jesseeichar** [comment]
I just had this issue as well.  As mentioned updating the configuration fixed it.  I am going to monitor and see if we continue to see this issue.  Maybe it only happens after unsealing or... If we get more information we will let you know.

**Comment by tstoermer** [comment]
Thanks for your information. We are also now on 0.11.1  . Until now, the issue did not reoccur.

Recently, there was a fix provided for context handling #13 
But I am not sure, if this could also solve this issue.
